src/java.base/share/classes/sun/security/ssl/AlpnExtension.java

```diff
@@ -107,7 +109,7 @@ private AlpnSpec(HandshakeContext hc,
                     "incorrect list length (length=" + listLen + ")"));
             }
 
-            List<String> protocolNames = new LinkedList<>();
+            Stream.Builder<String> alpnNames = Stream.builder();
```

Comment:
Did you have some benchmark numbers in this context about the performance improvement by replacing LinkedList with Stream.builder?

Reply:
Not at the moment but I'm working on setting up some JMH benchmarks now.